### PR TITLE
Add Ergosaurus Support

### PR DESCRIPTION
**Pull request by cfbender**
## Description

Wanting to add the Ergosaurus PCB, which should be available in the future through KeyHive. PCB was designed by /u/reggatronics.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/ergosaurus/ergosaurus.c [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "ergosaurus.h"
+
+// Optional override functions below.
```
Comment (noroadsleft) [inline keyboards/ergosaurus/ergosaurus.c]: You can delete Lines 19-52.

File: keyboards/ergosaurus/keymaps/default/readme.md [hunk 2]
```diff
@@ -0,0 +1 @@
+# The default keymap for reggalicious-alice
```
Comment (noroadsleft) [inline keyboards/ergosaurus/keymaps/default/readme.md]:
```suggestion
# The default keymap for ergosaurus
```

File: keyboards/ergosaurus/readme.md [hunk 3]
```diff
@@ -0,0 +1,15 @@
+# ergosaurus
+
+![ergosaurus](imgur.com image replace me!)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]: There should be an image link here.

File: keyboards/ergosaurus/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+A short description of the keyboard/project
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]: And a description here.

File: keyboards/ergosaurus/readme.md [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+
+Keyboard Maintainer: [cfbender](https://github.com/cfbender)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]:
```suggestion
* Keyboard Maintainer: [cfbender](https://github.com/cfbender)
```

File: keyboards/ergosaurus/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# ergosaurus
+
+![ergosaurus](imgur.com image replace me!)
+
+A short description of the keyboard/project
+
+Keyboard Maintainer: [cfbender](https://github.com/cfbender)  
+Hardware Supported: The PCBs, controllers supported
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]:
```suggestion
* Hardware Supported: Ergosaurus PCB, Pro Micro
```

File: keyboards/ergosaurus/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
+# ergosaurus
+
+![ergosaurus](imgur.com image replace me!)
+
+A short description of the keyboard/project
+
+Keyboard Maintainer: [cfbender](https://github.com/cfbender)  
+Hardware Supported: The PCBs, controllers supported  
+Hardware Availability: links to where you can find this hardware
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]: And a product link here (or even just a link to the KeyHive site).

File: keyboards/ergosaurus/rules.mk [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+# If you don't know the bootloader type, then you can specify the
+# Boot Section Size in *bytes* by uncommenting out the OPT_DEFS line
+#   Teensy halfKay      512
+#   Teensy++ halfKay    1024
+#   Atmel DFU loader    4096
+#   LUFA bootloader     4096
+#   USBaspLoader        2048
+# OPT_DEFS += -DBOOTLOADER_SIZE=4096
+
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]: Lines 14-22 should be deleted.

File: keyboards/ergosaurus/rules.mk [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite      # Virtual DIP switch configuration(+1000)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
```

File: keyboards/ergosaurus/rules.mk [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys(+4700)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
MOUSEKEY_ENABLE = yes       # Mouse keys
```

File: keyboards/ergosaurus/rules.mk [hunk 11]
```diff
@@ -0,0 +1,42 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control(+450)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
EXTRAKEY_ENABLE = yes       # Audio control and System control
```

File: keyboards/ergosaurus/rules.mk [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+CONSOLE_ENABLE = yes        # Console for debug(+400)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
```

File: keyboards/ergosaurus/rules.mk [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+COMMAND_ENABLE = yes        # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support (+2400 to 4200, depending on config)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
MIDI_ENABLE = no            # MIDI support
```

File: keyboards/ergosaurus/rules.mk [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

File: keyboards/ergosaurus/readme.md (hunk 3 above)
Comment (cfbender) [reply keyboards/ergosaurus/readme.md]: So strange, I know I added this - must've forgot to hit save! I'll get that pushed up today, thanks for catching!

File: keyboards/ergosaurus/keymaps/default/config.h [hunk 15]
```diff
@@ -0,0 +1,19 @@
+/* Copyright 2019 cfbender
```
Comment (fauxpark) [inline keyboards/ergosaurus/keymaps/default/config.h]: This file can be deleted.

File: keyboards/ergosaurus/readme.md [hunk 16]
```diff
@@ -0,0 +1,15 @@
+# Ergosaurus
+
+![ergosaurus](https://i.imgur.com/9yLCcpH.jpg)
+
+An Arduino Pro Micro driven Alice-clone PCB.
+
+* Keyboard Maintainer: [cfbender](https://github.com/cfbender)
```
Comment (fauxpark) [inline keyboards/ergosaurus/readme.md]:
```suggestion
* Keyboard Maintainer: [cfbender](https://github.com/cfbender)
```

File: keyboards/ergosaurus/readme.md [hunk 17]
```diff
@@ -0,0 +1,15 @@
+# Ergosaurus
+
+![ergosaurus](https://i.imgur.com/9yLCcpH.jpg)
+
+An Arduino Pro Micro driven Alice-clone PCB.
+
+* Keyboard Maintainer: [cfbender](https://github.com/cfbender)  
+* Hardware Supported: All Alice-style cases, given the clearance for the Pro Micro. Ergosaurus specific acrylic case files available [here](https://github.com/cfbender/Lubrigante)
```
Comment (fauxpark) [inline keyboards/ergosaurus/readme.md]:
```suggestion
* Hardware Supported: All Alice-style cases, given the clearance for the Pro Micro. Ergosaurus specific acrylic case files available [here](https://github.com/cfbender/Lubrigante)
```

File: keyboards/ergosaurus/config.h [hunk 18]
```diff
@@ -0,0 +1,252 @@
...
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID 0xBEEF
+#define PRODUCT_ID 0x0000
```
Comment (drashna) [inline keyboards/ergosaurus/config.h]: ✅  I have no beef with this VID/PID

Though, you may want to assign a non-zero PID here, too.

File: keyboards/ergosaurus/readme.md [hunk 19]
```diff
@@ -0,0 +1,15 @@
+# Ergosaurus
+
+![ergosaurus](https://i.imgur.com/9yLCcpH.jpg)
+
+An Arduino Pro Micro driven Alice-clone PCB.
+
+* Keyboard Maintainer: [cfbender](https://github.com/cfbender)
+* Hardware Supported: All Alice-style cases, given the clearance for the Pro Micro. Ergosaurus specific acrylic case files available [here](https://github.com/cfbender/Lubrigante)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]:
```suggestion
* Hardware Supported: Ergosaurus PCB, Pro Micro
```

File: keyboards/ergosaurus/readme.md [hunk 20]
```diff
@@ -0,0 +1,15 @@
+# Ergosaurus
+
+![ergosaurus](https://i.imgur.com/9yLCcpH.jpg)
+
+An Arduino Pro Micro driven Alice-clone PCB.
```
Comment (noroadsleft) [inline keyboards/ergosaurus/readme.md]:
```suggestion
An Arduino Pro Micro driven Alice-clone PCB. Supports all Alice-style cases, given the clearance for the Pro Micro. Ergosaurus specific acrylic case files available [here](https://github.com/cfbender/Lubrigante).
```

File: keyboards/ergosaurus/rules.mk [hunk 21]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no 		# Enable support for HD44780 based LCDs (+400)
```
Comment (noroadsleft) [inline keyboards/ergosaurus/rules.mk]: This is still pending from my previous review.

```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

Comment (cfbender) [reply keyboards/ergosaurus/rules.mk]: sorry, this wasn't showing up in my editor. hopefully I got it this time!

File: keyboards/ergosaurus/rules.mk [hunk 22]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+HD44780_ENABLE = no         # Enable support for HD44780 based LCDs (+400)
```
Comment (fauxpark) [inline keyboards/ergosaurus/rules.mk]:
```suggestion
HD44780_ENABLE = no         # Enable support for HD44780 based LCDs
```

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**